enginetest/evaluation.go

```diff
@@ -610,8 +611,12 @@ func rowToSQL(s sql.Schema, expectedRow sql.Row, actualRow sql.Row, convertTime
 		_, isStr := v.(string)
 		if v == nil {
 			newRow = append(newRow, nil)
-		} else if types.IsDecimal(s[i].Type) || (isTime && !convertTime) || (isStr && types.IsTextOnly(s[i].Type)) {
+		} else if types.IsDecimal(s[i].Type) || s[i].Type.Type() == sqltypes.Year || (isTime && !convertTime) || (isStr && types.IsTextOnly(s[i].Type)) {
 			newRow = append(newRow, v)
+		} else if types.IsEnum(s[i].Type) || types.IsSet(s[i].Type) {
+			// enginetests returns the enum index, but the text representation is sent over the wire
+			// TODO: need to check result rather than replacing it
```

Comment:
How much more glue do we need before we can convert the set/enum index into the string value to check against the expected result?

Reply:
In order to check the actual result against the expected for these tests, I think the best way is to use the actual value rather than use the index because there is not really any way to get the elements for the enum/set type from the schema for server engine tests. Instead, the elements should be accessible for regular engine tests, so we can use it to convert the returned index to actual string value in the `evaluation`